Python/Product/Analysis/Interpreter/InterpreterConfiguration.cs

```diff
@@ -55,7 +61,28 @@ public sealed class InterpreterConfiguration : IEquatable<InterpreterConfigurati
         /// <summary>
         /// Gets a friendly description of the interpreter
         /// </summary>
-        public string Description { get; }
+        public string Description => _fullDescription ?? _description;
+
+        /// <summary>
+        /// Changes the description to be less likely to be
+        /// ambiguous with other interpreters.
+        /// </summary>
+        private void SwitchToFullDescription() {
+            bool hasVersion = _description.Contains(Version.ToString());
+            bool hasArch = _description.IndexOf(Architecture.ToString(), StringComparison.CurrentCultureIgnoreCase) >= 0 ||
+                _description.IndexOf(Architecture.ToString("x"), StringComparison.CurrentCultureIgnoreCase) >= 0;
+
+            if (hasVersion && hasArch) {
+                // Regular description is sufficient
+                _fullDescription = null;
+            } else if (hasVersion) {
+                _fullDescription = "{0} ({1})".FormatUI(_description, Architecture);
+            } else if (hasArch) {
+                _fullDescription = "{0} ({1})".FormatUI(_description, Version);
+            } else {
+                _fullDescription = "{0} ({1}, {2})".FormatUI(_description, Version, Architecture);
```

Comment:
These will look strange if `_description` is empty, but looking at the code PythonRegistrySearch, it doesn't look like that happens in practice.